Tests/GRPCNIOTransportHTTP2Tests/Test Utilities/CertificateChain.swift

```diff
@@ -0,0 +1,316 @@
+/*
+ * Copyright 2025, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Crypto
+import Foundation
+import SwiftASN1
+import X509
+
+/// Create a certificate chain with a root and intermediate certificate stored in a trust roots file and
+/// key / certificate pairs for a client and a server. These can be used to establish mTLS connections
+/// with local trust.
+///
+/// Usage example:
+/// ```
+/// // Create a new certificate chain
+/// let certificateChain = try CertificateChain()
+/// // Tag our certificate files with the function name
+/// let fileNames = try certificateChain.writeToTemp(fileTag: #function)
+/// // Access the file paths of the certificate files.
+/// let clientCertPath = fileNames[.clientCert]!
+/// ...
+/// ```
+struct CertificateChain {
+  /// Each node in the chain has a certificate and a key
+  struct CertificateKeyPair {
+    let certificate: Certificate
+    let key: Certificate.PrivateKey
+  }
+
+  /// Leaf certificates can authenticate either a client or a server
+  enum Authenticating {
+    case client
+    case server
+  }
+
+  /// Writing the files to disk returns a dictionary with these keys to access the file locations
```

Comment:
Rather than a dictionary can we just have a `struct` with a property for each of these? We can avoid the `!` when accessing them each time then.